plugins/modules/keycloak_clientsecret_info.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2022, Fynn Chen <ethan.cfchen@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_clientsecret_info
+
+short_description: Retrieve client secret via Keycloak API
+
+version_added: 6.1.0
+
+description:
+  - This module allows you to get a Keycloak client secret via the Keycloak
+    REST API. It requires access to the REST API via OpenID Connect; the user
+    connecting and the client being used must have the requisite access rights.
+    In a default Keycloak installation, admin-cli and an admin user would work,
+    as would a separate client definition with the scope tailored to your needs
+    and a user having the expected roles.
+
+  - When retrieving a new client secret, where possible provide the client's
+    id (not client_id) to the module. This removes a lookup to the API to
+    translate the client_id into the client ID.
+
+options:
+  realm:
+    type: str
+    description:
+      - They Keycloak realm under which this client resides.
+    default: 'master'
+
+  id:
+    description:
+      - The unique identifier for this client.
+      - This parameter is not required for getting or generating a client secret but
+        providing it will reduce the number of API calls required.
+    type: str
+
+  client_id:
+    description:
+      - The client_id of the client. Passing this instead of id results in an
+        extra API call.
+    aliases:
+      - clientId
+    type: str
+
+
+extends_documentation_fragment:
+  - community.general.keycloak
+
+author:
+  - Fynn Chen (@fynncfchen)
+  - John Cant (@johncant)
+'''
+
+EXAMPLES = '''
+- name: Get a Keycloak client secret, authentication with credentials
+  community.general.keycloak_clientsecret_info:
+    id: '9d59aa76-2755-48c6-b1af-beb70a82c3cd'
+    realm: MyCustomRealm
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+  delegate_to: localhost
```

Comment:
To show best practices, it's best to add `no_log: true` to all these tasks. Same for the other module.
```suggestion
  delegate_to: localhost
  no_log: true
```